quota/quota.go

```diff
@@ -67,6 +69,34 @@ type Spec struct {
 	User string
 }
 
+// Name returns a textual representation of the Spec. Names are constant and may be relied not to
+// change in the future.
+//
+// Names are created as follows:
+// * Global quotas are mapped to "global/read" or "global/write"
+// * Tree quotas are mapped to "trees/$TreeID/$Kind". E.g., "trees/10/read".
+// * User quotas are mapped to "users/$User/$Kind". E.g., "trees/10/read".
+func (s Spec) Name() string {
+	group := strings.ToLower(s.Group.String())
+	kind := strings.ToLower(s.Kind.String())
+	if s.Group == Global {
+		return fmt.Sprintf("%v/%v", group, kind)
+	}
+	var user string
+	switch s.Group {
+	case Tree:
+		user = fmt.Sprint(s.TreeID)
+	case User:
+		user = s.User
+	}
+	return fmt.Sprintf("%vs/%v/%v", group, user, kind)
+}
+
+// String returns a description of Spec.
+func (s Spec) String() string {
```

Comment:
It's a personal nit: a spec name is very specific thing, whereas its String() form isn't necessarily the same (it could be more descriptive, for example).